nodejs/eks/examples/cluster/index.ts

```diff
@@ -26,6 +26,16 @@ const cluster2 = new eks.Cluster(`${projectName}-2`, {
     ],
 });
 
+// Create an EKS cluster with the fargate configuration.
+const vpc3 = new awsx.ec2.Vpc(`${projectName}-3`);
+const cluster3 = new eks.Cluster(`${projectName}-3`, {
+    vpcId: vpc3.id,
+    privateSubnetIds: vpc3.privateSubnetIds,
```

Comment:
Note that `vpcId` and `privateSubnetIds` are effectively necessary to use `fargate: true`:
1. Fargate for EKS only works in private subnets, and the default VPC has only public subnets, so you must always provide an explicit VPC along with `fargate: true`
2. The current design of this library requires passing both subnets and VPC in - ideally we would augment this library to accept an `awsx.ec2.Vpc` as an alternative to a `vpc.id`, and if that is passed, infer the public and private subnets from it.  That's an orthogonal change to this PR - so something we can do separately.

Reply:
> 1. Fargate for EKS only works in private subnets, and the default VPC has only public subnets, so you must always provide an explicit VPC along with fargate: true

We should call out the need for private subnets in the comments / docs.

> 2. The current design of this library requires passing both subnets and VPC in - ideally we would augment this library to accept an awsx.ec2.Vpc as an alternative to a vpc.id, and if that is passed, infer the public and private subnets from it. That's an orthogonal change to this PR - so something we can do separately.

SGTM 👍